component/client/chat/ChatHistory.tsx

```diff
@@ -7,51 +7,33 @@ import Message from "./Message";
 
 interface IHistoryProps {
   userID: string;
-  messages: IData[];
+  incomingMessage: IData;
   startPrivateChatWith: (userID: string) => void;
 }
 
 class ChatHistory extends React.Component<IHistoryProps, any> {
   private historyEnd: HTMLElement;
   private shouldScroll: boolean;
   private lastPosition: number;
+  private messages: JSX.Element[];
 
   constructor(props: any) {
     super(props);
     this.shouldScroll = true;
+    this.messages = [];
```

Comment:
Slight optimization when there are large amounts of messages being sent from firebase. Rather than dealing with concatenations, we will simply push all received messages onto the `message` array of `JSX.element`. This allows us to avoid reconstructing the same set of messages every time we receive a new message which is costly when we receive many messages at once. This reduces the lag of the component if 100+ messages are being sent.